### DISCUSSION TITLE
discord.Member

### DISCUSSION THREAD

**Discussion by haiderlikesrust** [Rapptz/discord.py#6884] [Q&A]
why cant we use discord.Member for people outside the server. like ?ban 123456789010 if the user it outside the server it wont ban him. why?

**Comment by esmaycat** [comment]
For banning someone who isn't in the server that you are invoking the command in, you can use either `discord.User` (which represents a user on discord) or (as the ban only really needs a snowflake) you can pass in a `discord.Object`. 

This is because `discord.Member` objects are attached to the guild that the member is in, therefore a user can have different `discord.Member` objects depending which server then are in. Obviously then, if you are attempting to use the `discord.Member` converter to ban someone who isn't in the current server it will not work since they do not have a `discord.Member` object relating to the server you are invoking the command in.

**Comment by Rapptz** [comment]
This is because `discord.Member` represents a user *in* a server. If you want to ban a member outside of the server as a typehint for the command, then you should use `discord.User` instead.

**Reply by haiderlikesrust** [reply]
Ok thanks but does `discord.User` everything like `discord.Member` I mean all functions stuff.

**Reply by Rapptz** [reply]
You can check the difference in the documentation, for [`UserConverter`](https://discordpy.readthedocs.io/en/latest/ext/commands/api.html#discord.ext.commands.UserConverter) and [`MemberConverter`](https://discordpy.readthedocs.io/en/latest/ext/commands/api.html#discord.ext.commands.MemberConverter).
